src/pipeline/error.rs

```diff
@@ -0,0 +1,32 @@
+use std::fmt::Debug;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("failed to decode message: {0}")]
+    DecodeError(#[from] prost::DecodeError),
+    #[error("failed to handle serde message: {0}")]
+    SerdeError(#[from] serde_json::Error),
+    #[error("failed to parse message: {0}")]
+    ParseError(#[from] std::str::Utf8Error),
+    #[error("problem communicating between threads: {0}")]
+    ThreadError(Box<dyn std::error::Error + Send + Sync>),
+}
+
+impl<T: Debug + Send + Sync + 'static> From<tokio::sync::mpsc::error::SendError<T>> for Error {
+    fn from(e: tokio::sync::mpsc::error::SendError<T>) -> Self {
+        Error::ThreadError(e.into())
+    }
+}
+impl From<tokio::sync::oneshot::error::RecvError> for Error {
+    fn from(e: tokio::sync::oneshot::error::RecvError) -> Self {
+        Error::ThreadError(e.into())
+    }
+}
+impl From<tokio::task::JoinError> for Error {
+    fn from(e: tokio::task::JoinError) -> Self {
+        Error::ThreadError(e.into())
+    }
+}
+
+pub type Res<T> = Result<T, Error>;
```

Comment:
makes sense. i went with `pub use error::Res` in `pipeline/mod.rs`, and went ahead and changed `Res` -> `Result` since I agree, we should follow standards